src/ImageProcessor.Web/Caching/DiskCache.cs

```diff
@@ -211,64 +211,62 @@ public override Task TrimCacheAsync()
+                            var count = files.Count;
+                            foreach (var fileInfo in files)
                             {
-                                // If the group count is equal to the max count minus 1 then we know we
-                                // have reduced the number of items below the maximum allowed.
-                                // We'll cleanup any orphaned expired files though.
-                                if (!this.IsExpired(fileInfo.CreationTimeUtc) && count <= MaxFilesCount - 1)
+                                if (token.IsCancellationRequested)
                                 {
                                     break;
                                 }
 
-                                // Remove from the cache and delete each CachedImage.
-                                CacheIndexer.Remove(fileInfo.Name);
-                                fileInfo.Delete();
-                                count--;
-                            }
-                            catch (Exception ex)
-                            {
-                                // Log it but skip to the next file.
-                                ImageProcessorBootstrapper.Instance.Logger.Log<DiskCache>($"Unable to clean cached file: {fileInfo.FullName}, {ex.Message}");
+                                try
+                                {
+                                    // If the group count is equal to the max count minus 1 then we know we have reduced the number of items below the maximum allowed
+                                    // We'll cleanup any orphaned expired files though
+                                    if (!this.IsExpired(fileInfo.CreationTimeUtc) && count <= MaxFilesCount - 1)
+                                    {
+                                        break;
+                                    }
+
+                                    // Remove from the cache and delete each CachedImage
+                                    CacheIndexer.Remove(fileInfo.Name);
+                                    fileInfo.Delete();
+                                    count--;
+                                }
+                                catch (Exception ex)
+                                {
+                                    // Log it but skip to the next file
+                                    ImageProcessorBootstrapper.Instance.Logger.Log<DiskCache>($"Unable to clean cached file: {fileInfo.FullName}, {ex.Message}");
+                                }
                             }
                         }
+                        catch (Exception ex)
+                        {
+                            // Log it but skip to the next directory
+                            ImageProcessorBootstrapper.Instance.Logger.Log<DiskCache>($"Unable to clean cached directory: {directory}, {ex.Message}");
```

Comment:
This is the same message as logged by `RecursivelyDeleteEmptyDirectories()`, but that won't cause any issues in debugging, as the exception message is also included.